crml/cennzx-spot/rpc/src/lib.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2020 Parity Technologies (UK) Ltd. and Centrality Investments Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Node-specific RPC methods for interaction with CENNZX.
+
+use std::{convert::TryInto, sync::Arc};
+
+use codec::Codec;
+use jsonrpc_core::{Error as RpcError, ErrorCode, Result};
+use jsonrpc_derive::rpc;
+use sp_api::ProvideRuntimeApi;
+use sp_arithmetic::traits::SaturatedConversion;
+use sp_blockchain::HeaderBackend;
+use sp_runtime::{generic::BlockId, traits::Block as BlockT, traits::SimpleArithmetic};
+
+pub use self::gen_client::Client as CennzxSpotClient;
+pub use crml_cennzx_spot_rpc_runtime_api::{
+	self as runtime_api, CennzxSpotApi as CennzxSpotRuntimeApi, CennzxSpotResult,
+};
+
+/// Contracts RPC methods.
+#[rpc]
+pub trait CennzxSpotApi<AssetId, Balance> {
```

Comment:
We shouldn't need to define a trait twice. Either this or CennzxSpotApi in runtime-api/src/lib.rs could be reused.